pkg/controller/flightctl/flightctl.go

```diff
@@ -54,45 +50,92 @@ type FlightCtlManager struct {
 	recorder     events.Recorder
 
 	flightctlClient         flightctlClient
+	flightctlServer         string
 	agentRegistrationServer string
 
 	cachedToken string
 }
 
-func (f *FlightCtlManager) ApplyResources(ctx context.Context) error {
-	var err error
+// StartReconcileFlightCtlResources starts a loop to continuously reconcile FlightCtl resources
+func (f *FlightCtlManager) StartReconcileFlightCtlResources(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Minute)
+	defer ticker.Stop()
+
+	// Helper function to apply resources and record errors
+	applyFunc := func() {
+		if err := f.ensureFlightCtlServer(); err != nil {
+			f.recorder.Event("FlightCtlServerFailed",
+				fmt.Sprintf("Failed to ensure FlightCtl server: %v", err))
+			return
+		}
 
-	// Apply kubernetes resources
-	err = f.applyKuberentesResources(ctx)
-	if err != nil {
-		return err
-	}
+		// Apply kubernetes resources
+		if err := f.applyKuberentesResources(ctx); err != nil {
+			f.recorder.Event("KubernetesResourcesFailed",
+				fmt.Sprintf("Failed to apply Kubernetes resources: %v", err))
+			return
+		}
 
-	// Create Repository resources
-	err = f.applyRepository(ctx)
-	if err != nil {
-		return err
+		// Create Repository resources
+		if err := f.applyRepository(ctx); err != nil {
+			f.recorder.Event("RepositoryFailed",
+				fmt.Sprintf("Failed to apply Repository resources: %v", err))
+			return
+		}
+
+		// Record successful sync
+		f.recorder.Event("ResourcesSynced", "Successfully synced FlightCtl resources")
 	}
 
-	return nil
+	// Run immediately for the first time
+	applyFunc()
+
+	// Then run every 5 minutes
+	for {
+		select {
+		case <-ticker.C:
+			applyFunc()
+		case <-ctx.Done():
+			return
+		}
+	}
 }
 
-func (f *FlightCtlManager) applyKuberentesResources(ctx context.Context) error {
-	var err error
+// If the flightctl server is not set, then:
+// 1. list all services in the cluster scope, find the one with name "flightctl-api", and set the server to the cluster IP of the service.
+// 2. if not found, return an error.
+// TODO: @xuezhaojun, this is a temporary solution, because in this release, `flightctl-server` is not pass by installer, will remove
+// this ensureFlightCtlServer() after `flightctl-server` is passed by installer.
+func (f *FlightCtlManager) ensureFlightCtlServer() error {
+	if f.flightctlServer != "" {
+		return nil
+	}
 
-	// Check if the FlightCtl namespace exists
-	_, err = f.clientHolder.KubeClient.CoreV1().Namespaces().Get(ctx, FlightCtlNamespace, metav1.GetOptions{})
+	services, err := f.clientHolder.KubeClient.CoreV1().Services("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
this could be quite expensive to list all services. Why call list instead of using lister?

Reply:
Sure, we can change to Lister.